system/logaccess/ElasticStack/ElasticStackLogAccess.hpp

```diff
@@ -73,13 +74,21 @@ class ELASTICSTACKLOGACCESS_API ElasticStackLogAccess : public CInterfaceOf<IRem
     const IPropertyTree * getTimestampTypeFormat(const char * indexpattern, const char * fieldname);
     const IPropertyTree * performAndLogESRequest(Client::HTTPMethod httpmethod, const char * url, const char * reqbody, const char * logmessageprefix, LogMsgCategory reqloglevel, LogMsgCategory resploglevel);
 
+    void esSearchMetaData(std::string & search, const LogAccessReturnColsMode retcolmode, const  StringArray & selectcols, unsigned size, offset_t from);
+    void getMinReturnColumns(std::string & columns);
+    void getDefaultReturnColumns(std::string & columns);
+    void getAllColumns(std::string & columns);
 public:
     ElasticStackLogAccess(const std::vector<std::string> &hostUrlList, IPropertyTree & logAccessPluginConfig);
     virtual ~ElasticStackLogAccess() override = default;
 
+    void createQueryString(std::string & queryString, std::string & queryIndex, const LogAccessConditions & options);
```

Comment:
style/personal: the method name (and any method name createXX) suggests it will return something (the query string in this case), whereas queryIndex is an ancillary return value, so I think it would be better as:
```
    std::string createQueryString(const LogAccessConditions & options, std::string & queryIndex);
```
or even:
```
    std::pair<std::string, std::string> createQueryString(const LogAccessConditions & options);
```

Reply:
yeah I see your point, will rename to match its functionality